### DISCUSSION TITLE
expand on pb.authStore.model?

### DISCUSSION THREAD

**Discussion by Phillor** [pocketbase/pocketbase#1824] [Q&A] [answered]
Hello there,

in the user collection I have a relation field that points to another collection called "profiles". While the user is logged in I would like to access the fields in the "profiles" collection for the logged in user but it seems like that I'm not able to expand the authStore.model for the "profiles" collection. Is this correct? Or maybe I'm making some kind of mistake.

If this isn't working the only alternative for this would be to query the "profiles" collection when logging in and saving it to a cookie so the profile data for the logged in user isn't getting lost when the page reloads, right?

Thank you in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
You can provide the expand query parameter to the `authWithPassword()` and `authRefresh()` calls as well:

```js
await pb.collection("users").authWithPassword("test@example.com", "1234567890", {}, {
    expand: "profile" // replace with your relation field name
})

await pb.collection("users").authRefresh({}, {
    expand: "profile" // replace with your relation field name
})
```

**Reply by Phillor** [reply]
Thank you for the answer. This really looks promising! Does this also work with `authWithOAuth2` ?
I mean it should for my understanding because it also takes a queryParam Object but it seems like that the `pb.authStore.model.expand.profile` is always `undefined`...

**Reply by ganigeorgiev** [reply]
Yes, it should work, but keep in mind that it is the **last parameter** and currently it is not very convenient:

```js
// see https://github.com/pocketbase/js-sdk#auth-handlers

await pb.collection('users').authWithOAuth2(provider, code, codeVerifier, redirectUrl, createData = {}, bodyParams = {}, queryParams = {});
```

(_I'm planning to soft-deprecate the multi-arguments form and provide an override function with just one config object, so that custom headers can also be set, but other tasks for now are with higher priority)_

Also make sure that the request satisfy your `profiles` collection "View API" rule.

If you are still not able to make it work, please export you collections structure (Admin UI > Settings > Export collections) and if possible provide a code sample/minimal repo how you are initializing the client and performing the authentication.

**Reply by Phillor** [reply]
You're right! My fault, my API rules were faulty. It works like a charm now. Thank you for the support :)

**Comment by Phillor** [comment]
Okay so it actually works as I described. But I have another thing that is somewhat related to my initial problem.

The reason why I created the profile collection is that I wanted to split up the information of the user. user collection -> sensitive user data. 
profile collection -> app relevant data which contains no personal data.

So when a profile, related to the user, exists, the expand queryParam on the authWithOAuth2 method works! But when the user first registers with my app there is no entry in the profile collection of course, which results in an error, trying to expand a non-existent relation when logging in.

I was trying to find a way to automatically create a profile when registering with the application but I was not able to find something like this. My question is now how this can be accomplished? Or are there other ways to solve this problem?

Thank you very much in advance!

**Reply by ganigeorgiev** [reply]
No, this is not possible at the moment via the web APIs. You'll have to send manually a second profile create request after the OAuth2 and call `authRefresh({}, { expand: "profile" })` afterward (_or update the model.expand manually with the result of the profile create, eg. `pb.authStore.model.expand.profile=resultOfProfileCreate`_).

At the moment, the `profile` record can be created automatically only if you [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and attach for example to the [`OnModelAfterCreate("users")` app event hook](https://pocketbase.io/docs/event-hooks/) (_or `OnRecordAfterAuthWithOAuth2Request("users")` if you want to target only the OAuth2 action_).

**Reply by Phillor** [reply]
Alright. Thank you for the support once again. It works as described by you. 👍
